### PR TITLE
Added ImageBasedUpgradeStage to clusterupgrade_types

### PR DESCRIPTION
**Pull request by eranco74**

### CODE REVIEW

File: api/v1alpha1/clusterupgrade_types.go [hunk 1]
```diff
@@ -38,14 +38,26 @@ type ImageBasedUpgrade struct {
...
+	ExtraManifests   ConfigMapRef           `json:"extraManifests,omitempty"`
+	RollbackTarget   string                 `json:"rollbackTarget,omitempty"`
 }
 
+// SeedImageRef defines the stage of the image based upgrade
+type ImageBasedUpgradeStage string
+
+const (
+	//UpgradePrep perform preliminary steps to get the node ready for image based upgrade
+	UpgradePrep ImageBasedUpgradeStage = "preUpgrgade"
```
Comment (imiller0) [inline api/v1alpha1/clusterupgrade_types.go:54]:
```suggestion
	UpgradePrep ImageBasedUpgradeStage = "upgradePrep"
```

File: api/v1alpha1/clusterupgrade_types.go [hunk 2]
```diff
@@ -38,14 +38,26 @@ type ImageBasedUpgrade struct {
...
+	//PreUpgrade backup the platform and application artifacts
+	PreUpgrade ImageBasedUpgradeStage = "preUpgrgade"
```
Comment (imiller0) [inline api/v1alpha1/clusterupgrade_types.go:56]:
```suggestion
	PreUpgrade ImageBasedUpgradeStage = "preUpgrade"
```

File: api/v1alpha1/clusterupgrade_types.go [hunk 3]
```diff
@@ -38,14 +38,26 @@ type ImageBasedUpgrade struct {
...
+	//Upgrade reboot to the new stateroot and reconfigure the cluster
+	Upgrade ImageBasedUpgradeStage = "preUpgrgade"
```
Comment (imiller0) [inline api/v1alpha1/clusterupgrade_types.go:58]:
```suggestion
	Upgrade ImageBasedUpgradeStage = "upgrade"
```

**Review by imiller0 — COMMENTED** [review]
There is overlap here with WIP posted under PR #5. Can you coordintate with @jc-rh on this update?

**Comment by eranco74** [comment]
Closing in favor of https://github.com/openshift-kni/lifecycle-agent/pull/5

**Pull request closed without merging**